WaterwitchGUI/src/main.cpp

```diff
@@ -0,0 +1,256 @@
+
+    glfwWindowHint(GLFW_CONTEXT_VERSION_MAJOR, 3);
+    glfwWindowHint(GLFW_CONTEXT_VERSION_MINOR, 3);
+    glfwWindowHint(GLFW_OPENGL_PROFILE, GLFW_OPENGL_CORE_PROFILE);
+    glfwWindowHint(GLFW_MAXIMIZED, true);
+    
+    GLFWwindow* window = glfwCreateWindow(800, 800, "Eastern Edge Waterwitch GUI", NULL, NULL);
+    if (!window)
+    {
+        glfwTerminate();
+        return -1;
+    }
+
+    glfwMakeContextCurrent(window);
+    glfwSwapInterval(1);
+
+    IMGUI_CHECKVERSION();
+    ImGui::CreateContext();
+    ImGuiIO& io = ImGui::GetIO();
+    ImGui::StyleColorsDark();
+
+    ImGui_ImplGlfw_InitForOpenGL(window, true);
+    ImGui_ImplOpenGL3_Init("#version 330");
+
+    while (!glfwWindowShouldClose(window))
+    {
+        glfwPollEvents();
+
+        ImGui_ImplOpenGL3_NewFrame();
+        ImGui_ImplGlfw_NewFrame();
+        ImGui::NewFrame();
+
+        //top menu bar
+        if (ImGui::BeginMainMenuBar()) {
+            if (ImGui::BeginMenu("Cameras")) {
+                if (ImGui::MenuItem("Open Camera Window")) {
+                    showCameraWindow = true;
+                }
+                ImGui::EndMenu();
+            }
+            if (ImGui::BeginMenu("Config")) {
+                if (ImGui::MenuItem("Open Config Editor")) {
+                    showConfigWindow = true;
+                }
+                if (ImGui::BeginMenu("Load Config")) {
+                    for (const auto& configFile : recursive_directory_iterator("configs")) {
+                        if (ImGui::MenuItem(configFile.path().filename().string().c_str())) {
+                            ifstream configJson(string("configs/") + configFile.path().filename().string());
+                            json configData = json::parse(configJson);
+                            std::strncpy(config.cam1ip, configData["cameraIPs"][0].get<std::string>().c_str(), sizeof(config.cam1ip));
+                            std::strncpy(config.cam2ip, configData["cameraIPs"][1].get<std::string>().c_str(), sizeof(config.cam2ip));
+                            std::strncpy(config.cam3ip, configData["cameraIPs"][2].get<std::string>().c_str(), sizeof(config.cam3ip));
+                            std::vector<int> buttons = configData["buttons"].get<std::vector<int>>();
+                            config.buttonActions.clear();
+                            for (int i = 0; i < buttons.size(); i++) {
+                                config.buttonActions.push_back(static_cast<ButtonAction>(buttons[i]));
+                            }
+                            std::vector<int> axes = configData["axes"].get<std::vector<int>>();
+                            config.axisActions.clear();
+                            for (int i = 0; i < axes.size(); i++) {
+                                config.axisActions.push_back(static_cast<AxisAction>(axes[i]));
+                            }
+                        }
+                    }
+                    ImGui::EndMenu();
```

Comment:
Please integrate this with existing functionality (see `CameraStreamUrls` function in `ROS2/ROS_workspace/src/cli_app/src/cli.cpp`, which is already integrated with `profiles_manager.py`). 

This is not to say that you should keep the `CameraStreamUrls` function as-is, in fact it should be probably split into two separate functions where one handles communication with `profiles_manager.py` and the other is for the cli aspect. However, it's not standard to completely rewrite existing functionality.